src/main/java/seedu/revised/command/subject/ReturnSubjectCommand.java

```diff
@@ -72,28 +72,30 @@ public void goToSubject(Subject subject) {
                 //} catch (IOException e) {
                 //Ui.printWritingError();
 
-            } catch (NumberFormatException e) {
-                Ui.printIndexError();
             } catch (NoSubjectException | NoTopicException e) {
-                Ui.printNoTopicError();
+                System.out.println(e.getMessage());
             } catch (RepeatedSubjectException e) {
-                Ui.printRepeatedTopicError();
-            } catch (IndexOutOfBoundsException e) {
-                Ui.printOutOfBoundsError();
+                System.out.println(e.getMessage());
             } catch (TaskTodoException e) {
-                Ui.printTodoError();
+                System.out.println(e.getMessage());
```

Comment:
We could create a method in Ui such as printError(Throwable err) and pass the error to that method for it to print out. In that way we let Ui handle all the printing and improves cohesion.